python/psautohint/ufoFont.py

```diff
@@ -831,12 +831,19 @@ def addComponent(self, baseGlyphName, transformation, identifier=None,
 
         for i, v in enumerate(transformation):
             if transformation[i] != self.DEFAULT_TRANSFORM[i]:
-                self.data.append(str(round(v, 9)))
+                self.data.append(str(self.norm_float(round(v, 9))))
 
         self.data.append("w%s" % self.width)
         glyph = self.glyphset[baseGlyphName]
         glyph.drawPoints(self)
 
+    @staticmethod
+    def norm_float(value):
+        """Converts a float (whose decimal part is zero) to integer"""
+        if isinstance(value, float) and value % 1 == 0:
+            return int(value)
+        return value
+
```

Comment:
While at it, can you make this a function and use it elsewhere in this file where we do:
```
if pos % 1 == 0:                                                    
    pos = int(pos)
```